lib/Myriad/Transport/Redis.pm

```diff
@@ -444,18 +441,27 @@ async method redis_from_pool {
 }
 
 method return_redis_to_pool ($instance) {
-    $log->tracef('Returning instance to pool, count now %d', 0 + $redis_pool->@*);
-    push $redis_pool->@*, $instance
+    if ((my $pool_count = 0 +  $redis_pool->@*) <= $max_pool_count) {
+        $log->tracef('Returning instance to pool, count now %d', 0 + $pool_count);
+        push $redis_pool->@*, $instance
+    } else {
+        $log->warnf('Max pool count reached! removing instance from pool, count now %d', $pool_count);
+        $self->remove_child($instance);
+    }
 }
 
 async method xreadgroup (@args) {
     my $instance = await $self->redis_from_pool;
-    # This should also be possible with a try/finally combination...
-    # but that's currently failing with the $redis_pool slot not being
-    # defined
```

Comment:
what do we mean by this? I don't think its still applicable

Reply:
it means that we shouldn't need the `->on_ready` sub, and instead would be able to write:

```
my $instance = await $self->redis_from_pool;
try {
 ...
} finally {
 $self->return_redis_to_pool($instance);
}
```

except we can't due to limitations in try/finally, Object::Pad and/or async/await.

Reply:
oh I see, yep makes sense now. I will put back the comment since its still valid.